docs/fundamentals/ai-for-software-engineering/github-copilot.md

```diff
@@ -0,0 +1,136 @@
+# Sign up & Working with GitHub Copilot
+
+GitHub Copilot is an AI-powered coding assistant that works inside editors like Visual Studio Code (VS Code). It helps by writing code suggestions, generating boilerplate, and even drafting tests. If you’re brand new to coding tools, this guide will walk you through each step.
+
+---
+
+## Getting started with Copilot in VS Code
+
+1. **Download and Install VS Code**  
+      - Go to [https://code.visualstudio.com/](https://code.visualstudio.com/) and download the version for your operating system (Windows, macOS, or Linux).  
+      - Install it using the installer.  
+
+2. **Open the Extensions view**  
+      - In VS Code, look at the left-hand sidebar. Click on the square-shaped icon (Extensions).  
+      - In the search bar at the top of the Extensions panel, type `GitHub Copilot`.  
+      - Click **Install**.  
+
+      ![VS Code extension marketplace with Copilot highlighted](../../assets/images/Github%20copilot%20extension%20download.png)
+
+3. **Sign in with GitHub**  
+      - After installation, a pop-up will ask you to sign in.  
+      - Click the sign-in button → a browser window will open → log in with your GitHub account.  
+      - Free if you’re a student or part of GitHub’s verified programs; otherwise, you may need a subscription.  
+   
+      ![GitHub login popup in VS Code](../../assets/images/login%20to%20GH%20from%20VS%20code.png)
+
+
+4. **Verify that Copilot works**  
+      - Open any file (for example, a `.py` or `.js` file).  
+      - Start typing a simple function like `def hello():`.  
+      - If Copilot is working, you’ll see a suggestion appear faintly in your editor.  
+
+---
+
+## Configuring Copilot (settings.json)
+
+Sometimes Copilot can feel “too noisy” with automatic completions. You can control this.
+
+1. In VS Code, **open the Command Palette** (Press `Ctrl+Shift+P` (Windows/Linux) or `Cmd+Shift+P` (Mac)).  
+
+2. Type `Preferences: Open User Settings (JSON)` and press Enter.  
+
+3. In the JSON file that opens, append this snippet:  
+
+   ```json
+   {
+     "editor.inlineSuggest.enabled": false,
+     "github.copilot.enable": {
+       "*": true
+     }
+   }
+   ```
+  ![VS Code Command Palette with 'settings.json' option](../../assets/images/GH%20settings.png)
+
+   - This disables automatic inline suggestions but keeps Copilot active.  
+   - You can trigger Copilot manually using `Ctrl+Enter` (Windows/Linux) or `Cmd+Enter` (Mac).  
+
+---
+
+**Troubleshooting tips**  
+
+   - If you don’t see `settings.json`: make sure you opened “Preferences: Open User Settings (JSON)” and not the regular Settings UI.  
+   - If Copilot suggestions never appear, check:  
+   - Your GitHub account has Copilot access (visit [https://github.com/features/copilot](https://github.com/features/copilot)).  
+   - The file type you’re editing is supported (e.g., `.py`, `.js`, `.ts`, `.java`).  
+
+---
```

Comment:
Tips like these, can be put into an admonition. See https://squidfunk.github.io/mkdocs-material/reference/admonitions/